resource-aggregate/service/aggregate.go

```diff
@@ -21,30 +21,66 @@ type Aggregate struct {
+		if aggregateID == resID.ToUUID().String() {
+			return resourceLinks, nil
+		}
+		return resourceState, nil
 	}
 }
 
-func NewResourceLinksFactoryModel(userID, owner, hubID string) func(ctx context.Context) (cqrsAggregate.AggregateModel, error) {
-	return func(context.Context) (cqrsAggregate.AggregateModel, error) {
+func NewResourceLinksFactoryModel(userID, owner, hubID string) func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
+	return func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
 		return events.NewResourceLinksSnapshotTakenForCommand(userID, owner, hubID), nil
 	}
 }
 
-func NewDeviceMetadataFactoryModel(userID, owner, hubID string) func(ctx context.Context) (cqrsAggregate.AggregateModel, error) {
-	return func(context.Context) (cqrsAggregate.AggregateModel, error) {
+func NewDeviceMetadataFactoryModel(userID, owner, hubID string) func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
+	return func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
 		return events.NewDeviceMetadataSnapshotTakenForCommand(userID, owner, hubID), nil
 	}
 }
 
-func NewServicesMetadataFactoryModel(userID, owner, hubID string) func(ctx context.Context) (cqrsAggregate.AggregateModel, error) {
-	return func(context.Context) (cqrsAggregate.AggregateModel, error) {
+func NewServicesMetadataFactoryModel(userID, owner, hubID string) func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
+	return func(context.Context, string, string) (cqrsAggregate.AggregateModel, error) {
 		return events.NewServiceMetadataSnapshotTakenForCommand(userID, owner, hubID), nil
 	}
 }
 
+func NewResourceAggregate(resourceID *commands.ResourceId, store eventstore.EventStore, factoryModel cqrsAggregate.FactoryModelFunc, retry cqrsAggregate.RetryFunc, addLinkedResources bool) (*Aggregate, error) {
+	a := &Aggregate{
+		eventstore: store,
+	}
+	addLink := make([]cqrsAggregate.AdditionalModel, 0, 1)
+	if addLinkedResources {
+		addLink = append(addLink, cqrsAggregate.AdditionalModel{
+			GroupID:     resourceID.GetDeviceId(),
+			AggregateID: commands.NewResourceID(resourceID.GetDeviceId(), commands.ResourceLinksHref).ToUUID().String(),
+		})
+	}
+
+	cqrsAg, err := cqrsAggregate.NewAggregate(resourceID.GetDeviceId(),
+		resourceID.ToUUID().String(),
+		retry,
+		store,
+		factoryModel,
+		func(string, ...interface{}) {
+			// no-op - we don't want to log debug/trace messages
+		},
+		// load also links state
+		addLink...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create aggregate for resource: %w", err)
+	}
+	a.ag = cqrsAg
+	return a, nil
+}
+
 // NewAggregate creates new resource aggreate - it must be created for every run command.
 func NewAggregate(resourceID *commands.ResourceId, store eventstore.EventStore, factoryModel cqrsAggregate.FactoryModelFunc, retry cqrsAggregate.RetryFunc) (*Aggregate, error) {
```

Comment:
I think this can just call `NewResourceAggregate` with the last parameter set to `false`

Reply:
fixed